packages/rum-core/src/performance-monitoring/capture-hard-navigation.js

```diff
@@ -25,72 +25,139 @@
   )
 }
 
 function createNavigationTimingSpans (timings, baseTime) {
-  var spans = []
-  for (var i = 0; i < eventPairs.length; i++) {
-    var start = timings[eventPairs[i][0]]
-    var end = timings[eventPairs[i][1]]
-    if (
-      baseTime &&
-      start &&
-      end &&
-      end > start &&
-      start >= baseTime &&
-      end - start < spanThreshold &&
-      start - baseTime < spanThreshold &&
-      end - baseTime < spanThreshold
-    ) {
-      var span = new Span(eventPairs[i][2], 'hard-navigation.browser-timing')
-      if (eventPairs[i][0] === 'requestStart') {
-        span.pageResponse = true
-      }
-      span._start = start - baseTime
-      span.ended = true
-      span._end = end - baseTime
-      spans.push(span)
+  const spans = []
+  for (let i = 0; i < eventPairs.length; i++) {
+    const start = timings[eventPairs[i][0]]
+    const end = timings[eventPairs[i][1]]
+
+    if (!isValidPerformanceTiming(start, end, baseTime)) {
+      continue
+    }
+    const span = new Span(eventPairs[i][2], 'hard-navigation.browser-timing')
+    if (eventPairs[i][0] === 'requestStart') {
+      span.pageResponse = true
     }
+    span._start = start - baseTime
+    span.ended = true
+    span._end = end - baseTime
+    spans.push(span)
   }
   return spans
 }
 
+function createResourceTimingSpan (name, initiatorType, start, end) {
+  let kind = 'resource'
+  if (initiatorType) {
+    kind += '.' + initiatorType
+  }
+
+  const parsedUrl = new Url(name)
+  const spanName = parsedUrl.origin + parsedUrl.path
+  const span = new Span(spanName || name, kind)
+  span.addContext({
+    http: {
+      url: name
+    }
+  })
+  span._start = start
+  span.ended = true
+  span._end = end
+  return span
+}
+
 function createResourceTimingSpans (entries, filterUrls) {
-  var spans = []
-  for (var i = 0; i < entries.length; i++) {
-    var entry = entries[i]
-    var { initiatorType, name, startTime, responseEnd } = entry
+  const spans = []
+  for (let i = 0; i < entries.length; i++) {
+    let { initiatorType, name, startTime, responseEnd } = entries[i]
     /**
      * Skipping the timing information of API calls because of auto patching XHR and Fetch
      */
-    if (initiatorType === 'xmlhttprequest' || initiatorType === 'fetch' || !entry.name) {
-      continue
-    } else if (
-      initiatorType !== 'css' &&
-      initiatorType !== 'img' &&
-      initiatorType !== 'script' &&
-      initiatorType !== 'link'
+    if (
+      initiatorType === 'xmlhttprequest' ||
+      initiatorType === 'fetch' ||
+      !name
     ) {
-      // is ajax request? test for css/img before the expensive operation
-      var foundAjaxReq = false
-      for (var j = 0; j < filterUrls.length; j++) {
-        var idx = name.lastIndexOf(filterUrls[j])
+      continue
+    } else if (!isValidPerformanceTiming(startTime, responseEnd)) {
+      continue
+    } else if (RESOURCE_INITIATOR_TYPES.indexOf(initiatorType) !== -1) {
```

Comment:
This doesn't have to be part of the else if. since the above ifs have continue in them you can start another if else block after.